### PR TITLE
feature/skip preliminary analysis on dia

### PR DESCRIPTION
**Pull request by jspaezp**
This PR adds the option to skip the preliminary steps of the dia analysis. (only do a single individual analysis and a single consensus analysis).

(please squash on merge ...)

### CODE REVIEW

**Comment by ypriverol** [comment]
Before reviewing the PR in details @jspaezp can you explain the impact of not doing preliminary step. I thought the idea with the preliminary step is to be able to generate the library for the final analysis @daichengxin ?

**Comment by jspaezp** [comment]
image

@ypriverol Absolutely!
1. The idea behind the feature is to allow two-stage runs of the pipeline, where a subset of files are used to generate the empirical library and then the extraction is done in all/the rest.
2. This is especially relevant because the empirical lib generation stage requires staging all the files (all .d + all .quant + fasta + predicted library) in the same compute environment/disk. (relevant discussion: https://twitter.com/J_my_sci/status/1744152837247095086)

**Comment by jspaezp** [comment]
btw... i dont believe any of the error in the ci/cd checks are caused by my changes ... I see a couple of files missing upstream and mamba not being able to generate environments.

**Comment by ypriverol** [comment]
We have to solve that, it is a work in progress because we have to move some files from the current server to PRIDE.

**Review by daichengxin — APPROVED** [review]

**Comment by jspaezp** [comment]
thanks @daichengxin for the review!

**Review by ypriverol — APPROVED** [review]

**Comment by ypriverol** [comment]
@jspaezp do you know which impact can have if you subselect a group of files compared to all the files in the final results?

**Comment by jspaezp** [comment]
I have not tested systematically this to be sure BUT. I would assume that (1) you could miss peptides that show up specifically in one of the conditions/files not used in the library construction. (2) You would have a slightly worse estimate of your FDR due to the smaller sample size.

I am not sure what public dataset could be used to test this hypothesis ... And I am assuming that data sets with more variability will be more prone to have changes depending on the analysis workflow. (I would be surprised if a 'cell line'+treatment dataset of 500 files looks any different if the library is done with the 500 files or with 100).

**Comment by ypriverol** [comment]
> I have not tested systematically this to be sure BUT. I would assume that (1) you could miss peptides that show up specifically in one of the conditions/files not used in the library construction. (2) You would have a slightly worse estimate of your FDR due to the smaller sample size.
> 
> I am not sure what public dataset could be used to test this hypothesis ... And I am assuming that data sets with more variability will be more prone to have changes depending on the analysis workflow. (I would be surprised if a 'cell line'+treatment dataset of 500 files looks any different if the library is done with the 500 files or with 100).

Do you think in the logic we can use some of the SDRF information to do this selection?

**Comment by jspaezp** [comment]
I was thinking about this for a while and there might be a way, but it would certainly require a lot more nexflow plumbing that I really want to/can afford to devote right now ... In addition, I am not sure what cvparam could be used to denote that those should be used for the lib ... 1002752 ?? maybe ?

We could certainly have it as an open issue to implement the feature in the future (we could also discuss the right way to do it in the issue).

In other words, that is a much more complex feature than this PR attempts to be and I believe this feature by itself is complementary to that one.
